### DISCUSSION TITLE
"Can't execute 'bash' error" on Alpine image

### DISCUSSION THREAD

**Discussion by amrishparmar** [docker-library/postgres#1217] [General]
I noticed recently that my deployment of the Alpine-based image stopped working correctly.

It's a simple repro:
```
$ docker run --rm postgres:16.2-alpine
env: can't execute 'bash': Operation not permitted
```

I tried it with 16.1 and 16.2 and they both fail in the same way.
I then tried some older 15.x releases and they all work correctly up until 15.4, but then from 15.5 onwards it starts failing with the above error. 16.0 also works correctly

Table of versions I've tried:

<table>
<tr>
 <th>Version
 <th>Working?
<tr>
 <td>15.3
 <td>:heavy_check_mark:
<tr>
 <td>15.4
 <td>:heavy_check_mark:
<tr>
 <td>15.5
 <td>:x:
<tr>
 <td>15.6
 <td>:x:
<tr>
 <td>16.0
 <td>:heavy_check_mark:
<tr>
 <td>16.1
 <td>:x:
<tr>
 <td>16.2
 <td>:x:

</table>

I don't see this error on the standard Debian images.

Docker version:
```
$ docker --version
Docker version 25.0.4, build 1a576c5
```

**Comment by tianon** [comment]
My best guess is that you need to update `libseccomp2` (and possibly, but less likely, `runc`) on your host :eyes:

**Reply by tianon** [reply]
To illustrate further (I can't reproduce :sweat_smile:):

```console
$ docker run --rm --pull=always postgres:16.2-alpine
16.2-alpine: Pulling from library/postgres
Digest: sha256:1d74239810c19ed0dbb317acae40974e673865b2d565a0d369e206159d483957
Status: Image is up to date for postgres:16.2-alpine
Error: Database is uninitialized and superuser password is not specified.
       You must specify POSTGRES_PASSWORD to a non-empty value for the
       superuser. For example, "-e POSTGRES_PASSWORD=password" on "docker run".

       You may also use "POSTGRES_HOST_AUTH_METHOD=trust" to allow all
       connections without a password. This is *not* recommended.

       See PostgreSQL documentation about "trust":
       https://www.postgresql.org/docs/current/auth-trust.html
```

**Reply by amrishparmar** [reply]
I'm no longer able to replicate as I had to reinstall my OS (due to a separate issue where I got the classic login loop problem).

I was on Ubuntu 22.04 LTS, but now on Ubuntu 23.10 and it seems to work correctly.
